### PR TITLE
smbios: cleaning up smbios requirements

### PR DESCRIPTION
**Pull request by david-sia**
Cleaning up SMBIOS requirements by removing duplicates found in the current specification and instead referencing the corresponding requirements outlined in the DMTF SMBIOS Specification.

### CODE REVIEW

File: smbios.adoc [hunk 1]
```diff
@@ -16,24 +16,23 @@ NOTE: The structures and fields in this section are defined in a manner consiste
 [%header, cols="5,25"]
 |===
 | ID#     ^| Requirement
-| `SMBIOS_010` | A Baseboard/Module Information (Type 02) structure SHOULD be implemented.
-| `SMBIOS_020` | A System Enclosure/Chassis (Type 03) structure SHOULD be implemented.
+| `SMBIOS_010` | The essential structures and data from Table 4, Section 6.2, along with the conformance guidelines in Annex A of the cite:[SMBIOS] specification, MUST be implemented if hardware support is present.
```
Comment (andreiw) [inline smbios.adoc:19]: Annex A is informative, i.e. non-normative, so let's instead refer to Annex A inside our own smbios guidelines and drop it from here

File: smbios.adoc [hunk 2]
```diff
@@ -16,24 +16,23 @@ NOTE: The structures and fields in this section are defined in a manner consiste
 [%header, cols="5,25"]
 |===
 | ID#     ^| Requirement
-| `SMBIOS_010` | A Baseboard/Module Information (Type 02) structure SHOULD be implemented.
```
Comment (andreiw) [inline smbios.adoc:19]: Why did you drop Type 02 here?

**Review by andreiw — CHANGES_REQUESTED** [review]
Take a look at the comments and please make the relevant changes.

File: non-normative/smbios.adoc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Note the DMTF requirements on the 64-bit SMBIOS 3.0 entry point (cite:[SMBIOS] Section 5.2.2) and data structures (cite:[SMBIOS] Section 6.2).
+Note the DMTF requirements on the 64-bit SMBIOS 3.0 entry point (cite:[SMBIOS] Section 5.2.2) and the SMBIOS requirement at <<smbios>>.
```
[the remark this answers is not in the document]
Comment (david-sia) [reply non-normative/smbios.adoc:1]: OK

File: smbios.adoc (hunk 1 above)
Comment (david-sia) [reply smbios.adoc:19]: noted

File: smbios.adoc (hunk 2 above)
Comment (david-sia) [reply smbios.adoc:19]: The reason the SMBIOS Table 02 requirement was omitted in this patch is that we have specified in SMBIOS_010 to adhere to DMTF SMBIOS requirements. Within SMBIOS_010, it is explicitly stated that these tables must be implemented if the hardware is supported. Therefore, I consider it unnecessary to keep it in the table.

**Comment by david-sia** [comment]
@andreiw, I have made the necessary changes. Please help to review again.

Thanks
Regards
Jee Heng

File: smbios.adoc [hunk 4]
```diff
@@ -16,24 +16,23 @@ NOTE: The structures and fields in this section are defined in a manner consiste
...
-| `SMBIOS_020` | A System Enclosure/Chassis (Type 03) structure SHOULD be implemented.
+| `SMBIOS_010` | The essential structures and data from Table 4, Section 6.2 of the cite:[SMBIOS] specification, MUST be implemented if hardware support is present.
```
Comment (andreiw) [inline smbios.adoc:19]: You're basically saying here that SMBIOS spec requirements should be honored. When working on ACPI and UEFI sections, this was a particular pain point. Originally we had followed the model set by SystemReady specs, which would duplicate requirements. Many on the BRS team felt this was confusing and unnecessary.

It should not be necessary to state this requirement. We already have "This section defines the BRS-I mandatory and optional SMBIOS requirements on top of existing cite:[SMBIOS] specification requirements." at the top.

File: non-normative/smbios.adoc [hunk 5]
```diff
@@ -1,4 +1,4 @@
-Note the DMTF requirements on the 64-bit SMBIOS 3.0 entry point (cite:[SMBIOS] Section 5.2.2) and data structures (cite:[SMBIOS] Section 6.2).
+Note the DMTF requirements on the 64-bit SMBIOS 3.0 entry point outlined in Section 5.2.2, as well as the conformance guidelines provided in Annex A of the cite:[SMBIOS] specification.
```
Comment (andreiw) [inline non-normative/smbios.adoc:1]: Note the DMTF requirements on the 64-bit SMBIOS 3.0 entry point (cite:[SMBIOS] Section 5.2.2) and the conformance guidelines (cite:[SMBIOS] Annex A).

The way you have it now suggests Section 5.2.2 is something in this document, while "of the cite:[SMBIOS] specification" might get rendered weird (of the [22] specification or similar).

**Review by andreiw — CHANGES_REQUESTED** [review]
A few more comments

File: smbios.adoc (hunk 4 above)
Comment (david-sia) [reply smbios.adoc:19]: Okay, point taken. I will put back the original statement of the SMBIOS_010.

File: non-normative/smbios.adoc (hunk 5 above)
Comment (david-sia) [reply non-normative/smbios.adoc:1]: Noted, will make the necessary changes.

**Comment by david-sia** [comment]
@andreiw, I have made the necessary changes. Please help to review again.

Thanks
Regards
Jee Heng

**Review by andreiw — APPROVED** [review]

**Pull request merged**